pkg/deploy/controller/deployment/controller.go

```diff
@@ -144,13 +144,18 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Generate a deployer pod spec.
 		deployerPod, err := c.makeDeployerPod(deployment)
 		if err != nil {
-			return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			nextStatus = deployapi.DeploymentStatusFailed
+			updatedAnnotations[deployapi.DeploymentStatusReasonAnnotation] = err.Error()
+			c.emitDeploymentEvent(deployment, kapi.EventTypeWarning, "FailedDecoding", err.Error())
+			break
 		}
 		// Create the deployer pod.
 		deploymentPod, err := c.pn.Pods(deployment.Namespace).Create(deployerPod)
-		// Retry on error.
 		if err != nil {
-			return actionableError(fmt.Sprintf("couldn't create deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
```

Comment:
> before this change, did the retry re-fetch the deploymentconfig as part of the retry (to make sure a deleted namespace or deploymentconfig doesn't eternally retry based on internal controller state)

Retries are bounded (in all deployment controllers). The deployment config is not fetched, but always decoded from the rc annotation. I was thinking of falling back to fetching in case of a decoding failure.